### PR TITLE
Record invoices cjm

### LINKED ISSUE #9: Record Endpoints - Invoices
- request tests

### PR DESCRIPTION
**Pull request by CjMoore**
- adds record endpoints for invoices
- adds find and find_all functionality for invoices
- finding by created_at/updated_at is impossible

closes #9

### CODE REVIEW

File: app/models/invoice.rb [hunk 1]
```diff
@@ -5,4 +5,24 @@ class Invoice < ApplicationRecord
   has_many :transactions
 
   validates :status, presence: true
+
+  def self.get_by_key(key, params)
```
Comment (CjMoore) [inline app/models/invoice.rb]: @s-espinosa My understanding is that you can put methods in your ApplicationRecord that can be shared across models. If this is the case, would you recommend putting these methods there?

File: spec/requests/api/v1/invoices/can_find_invoices_by_query_spec.rb [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+    get "/api/v1/invoices/find?status=#{invoice1.status}"
+
+    expect(response).to be_success
+
+    invoice = JSON.parse(response.body)
+    expect(invoice["id"]).to eq(invoice1.id)
+  end
+
+  # it "can find by created_at" do
```
Comment (CjMoore) [inline spec/requests/api/v1/invoices/can_find_invoices_by_query_spec.rb]: @s-espinosa We've been struggling a lot with being able to find by "created_at" and "updated_at". Any suggestions on how to get past this?

screen shot 2017-03-13 at 8 47 08 pm

File: app/models/invoice.rb (hunk 1 above)
Comment (s-espinosa) [reply app/models/invoice.rb]: Let's discuss tomorrow. Can you articulate to me the advantages/disadvantages of doing this?

File: spec/requests/api/v1/invoices/can_find_invoices_by_query_spec.rb (hunk 2 above)
Comment (s-espinosa) [reply spec/requests/api/v1/invoices/can_find_invoices_by_query_spec.rb]: Not positive what all you've tried, but my recommendation is to add the created at explicitly to the create in your test and then use that when you find. Let's talk about this tomorrow as well.

**Review by s-espinosa — COMMENTED** [review]
Good questions. See responses below. Thanks!

File: app/models/application_record.rb [hunk 3]
```diff
@@ -1,3 +1,26 @@
 class ApplicationRecord < ActiveRecord::Base
   self.abstract_class = true
+
+  def self.get_by_key(key, params)
```
Comment (TheKaterTot) [inline app/models/application_record.rb:4]: I haven't seen class methods in the application controller before. This is really convenient for our purposes, and I'm happy with them here. If we have time, we should research if they can go (or belong) somewhere else.

**Comment by TheKaterTot** [comment]
👍

**Pull request merged**